app/src/androidTest/java/com/PreProcessorTest.java

```diff
@@ -112,4 +123,46 @@ public void performTest() {
             assertTrue(photoIsCorrect(images[i], faces[i]));
         }
     }
+
+    @Test
+    /**
+     * Testing if face on image works correctly.
+     */
+    public void IsFaceOnImageTest(){
+        Bitmap faceNotInImagePhoto = BitmapFactory.decodeResource(res, R.drawable.face_not_in_image);
+        Bitmap faceInImagePhoto = BitmapFactory.decodeResource(res, R.drawable.face_in_image);
+
+        Task<List<Face>> faceOutsideImage = preProcessor.detectAllFacesUsingML(faceNotInImagePhoto);
+        Task<List<Face>> faceInsideImage = preProcessor.detectAllFacesUsingML(faceInImagePhoto);
+
+        preProcessor.waitForTask(faceOutsideImage);
+        preProcessor.waitForTask(faceInsideImage);
+
+        assertFalse(preProcessor.isFaceOnImage(faceOutsideImage.getResult().get(0), faceNotInImagePhoto));
+        assertTrue(preProcessor.isFaceOnImage(faceInsideImage.getResult().get(0), faceInImagePhoto));
+    }
+
+    @Test
+    /**
+     * Check rotate and trim face works correctly.
+     */
+    public void rotateAndTrimFace(){
+        Bitmap faceInImagePhoto = BitmapFactory.decodeResource(res, R.drawable.face_in_image);
+        Task<List<Face>> faceInsideImage = preProcessor.detectAllFacesUsingML(faceInImagePhoto);
+        preProcessor.waitForTask(faceInsideImage);
+
+        Bitmap trimmed = preProcessor.rotateAndTrimFace(faceInImagePhoto, faceInsideImage.getResult().get(0));
+        Bitmap expectedTrimmed = BitmapFactory.decodeResource(res, R.drawable.face_trimmed);
+        equals(trimmed,expectedTrimmed);
```

Comment:
This test also tests nothing and always passes. There is no assert.

Reply:
after adding assert this test will never pass due to loading bitmap difference, i will remove this test